### ISSUE TITLE
undefined parse transform 

### ISSUE THREAD

**Issue by Direnol** [erlang-ls/erlang_ls#446]
**Describe the bug**
Error using custom parse_transform.

**To Reproduce**
Open project _erlang_ls/priv/code_navigate_
  and open the _src / diagnostics_parse_transform_usage.erl_

**Expected behavior**
Issuing other errors and find parse_transform

**Actual behavior**
> undefined parse transform 'MODULE'

 Stop issuing other errors.

**Context**
 - `erlang_ls` 1a3d588
 - Editor used: vscode
 - LSP client used: erlang_ls

**Comment by onno-vos-dev** [comment]
I tried this out and weirdly enough it works if I open `diagnostics_parse_transform.erl`, save and switch back to `diagnostics_parse_transform_usage.erl` so something is funny here :) 

Maybe someone who's more familiar with parse_transform knows why this happens? On a project I work with which uses a custom parse_transform I do not have this problem though and it works out of the box. 

@Direnol btw, the version of `erlang_ls` that you refer to is old, seeing as you refer to the `diagnostics_parse_transform_usage.erl` file which was introduced in #442 I'm assuming the version you're referring to is outdated?

**Comment by Direnol** [comment]
Yeah,it's work now. But i have a case, where it don't work. This hrl is part of dependency
`-include_lib("chronica/include/chronica.hrl").`

```erlang
-ifndef(_CHRONICA_HRL_).
-define(_CHRONICA_HRL_, 1).
-compile({parse_transform, pt_chronica}).
-endif.
```

**Comment by robertoaloi** [comment]
@Direnol I would try by changing `deps_dirs` to:

    deps_dirs:
      - "deps/*"

The fact the server can't find the parse transform could indicate that dependency is not indexed correctly. I'm just guessing your project structure, though. If this does not help, please provide either a link to the project itself or a skeleton with the same structure, so we can troubleshoot this further.

**Comment by robertoaloi** [comment]
@Direnol Also, you can troubleshoot the issue by [looking at the server logs](https://github.com/erlang-ls/erlang_ls#troubleshooting). You should see a message stating the parse transform cannot be found.

**Comment by robertoaloi** [comment]
@onno-vos-dev The reason why you experience that behaviour for the `code_navigation` app is that it sits in a `priv` dir and the server picks the top-level "erlang_ls" as the project root. That application is only there for test purposes. If you want to navigate the `code_navigation` app you can either move the app to a different location or remove the top level `erlang_ls.config`.

**Comment by Direnol** [comment]
erlang_ls: 2de099a
Again i got 
> undefined parse transform 'diagnostics_parse_transform

```erlang
λ cat src/test_pt.erl 
-module(test_pt).
-author("Roman Mingazeev").

-export([]).
-include_lib("code_navigation/priv/code_navigation/include/parse_transform.hrl").
```

```elixir
#mix.exs
...
 defp deps do
    [
      {:code_navigation, git: "git@github.com:erlang-ls/erlang_ls.git", sparse: "priv/code_navigation/"}
    ]
  end

```


```yaml
#erlang_ls.config
apps_dirs:
  - "apps/*"

otp_apps_exclude:
  - "diameter"
  - "wx"
  - "snmp"

include_dirs:
  - "apps/*/include"
  - "deps/*/include"
  - "deps/"

deps_dirs:
  - "deps/*"

```

[debug.log](https://github.com/erlang-ls/erlang_ls/files/4234329/debug.log)

**Comment by robertoaloi** [comment]
The content of `priv` is not indexed by default, so the following will not work unless you explicitly ask the indexer to index that dir:

    -include_lib("code_navigation/priv/code_navigation/include/parse_transform.hrl").

**Comment by Direnol** [comment]
> The content of `priv` is not indexed by default, so the following will not work unless you explicitly ask the indexer to index that dir:
> 
> ```
> -include_lib("code_navigation/priv/code_navigation/include/parse_transform.hrl").
> ```

It's work. But this case is not work:

```erlang
% src/main.erl
-module(main).
-author("Roman Mingazeev").

-include_lib("parse_trans/include/exprecs.hrl").


-export([]).
```

```elixir
#mix.exs
...  defp deps do
    [
      {:parse_trans, git: "git@github.com:uwiger/parse_trans.git"}
    ]
  end
```

```yaml
# erlang_ls.config
apps_dirs:
  - "apps/*"

otp_apps_exclude:
  - "diameter"
  - "wx"
  - "snmp"

include_dirs:
  - "apps/*/include"
  - "deps/*/include"
  - "deps/"

deps_dirs:
  - "deps/*"
  - "deps/"
```
[crash.log](https://github.com/erlang-ls/erlang_ls/files/4239428/crash.log)
[debug.log](https://github.com/erlang-ls/erlang_ls/files/4239429/debug.log)
